### PR TITLE
[POAE7-2466]support VARCHAR data generation for arrow format

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
1. Support utf8 string for arrow format, able to generate string column randomly.
2. add toString function for arrow format.
3. resolve some comments.

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
add Unit Tests

### CODE REVIEW

File: cider/tests/utils/ArrowArrayBuilderTest.cpp [hunk 1]
```diff
@@ -122,54 +129,41 @@ TEST(ArrowArrayBuilderTest, OneColumnBatch) {
...
-//           .setRowNum(5)
-//           .addColumn<std::string>(
-//               "VarChar", CREATE_SUBSTRAIT_TYPE(Varchar), {"a", "b", "c", "d", "e"})
-//           .build();
-//   EXPECT_EQ(5, batch3.row_num());
-//   EXPECT_EQ(1, batch3.column_num());
-// }
+  EXPECT_EQ("u", std::string(schema->children[0]->format));
+
+  int32_t str1_start = *(int32_t*)(array->children[0]->buffers[1]);
```
Comment (jikunshang) [inline cider/tests/utils/ArrowArrayBuilderTest.cpp:143]: can we define a util function like `std::string ExtractUttf8ArrowArrayAt()`

File: cider/tests/utils/QueryArrowDataGenerator.h [hunk 2]
```diff
@@ -197,6 +208,47 @@ class QueryArrowDataGenerator {
...
+                              const int64_t min_len = 0,
+                              const int64_t max_len = -1) {
+    CHECK_GE(min_len, 0);
+    CHECK_GE(max_len, min_len);
+    std::string col_data = "";
+    std::vector<bool> null_data(row_num);
+    std::vector<int32_t> offset_data;
+    std::mt19937 rng(std::random_device{}());  // NOLINT
+    switch (pattern) {
+      case GeneratePattern::Sequence:
```
Comment (jikunshang) [inline cider/tests/utils/QueryArrowDataGenerator.h:225]: same for Sequence and Random?

**Review by jikunshang — COMMENTED** [review]
Over LGTM. Please edit your PR title with JIRA id.

File: cider/exec/module/batch/CiderBatchUtils.cpp [hunk 3]
```diff
@@ -339,4 +339,15 @@ std::unique_ptr<CiderBatch> createCiderBatch(std::shared_ptr<CiderAllocator> all
                   std::string("Unsupported data type to create CiderBatch: ") + format);
   }
 }
+
+std::string extractUttf8ArrowArrayAt(const ArrowArray* array, size_t index) {
```
Comment (yma11) [inline cider/exec/module/batch/CiderBatchUtils.cpp:343]: typo? should be `extractUttf8ArrowArrayAt`?

**Review by yma11 — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**